### PR TITLE
Fixed flat report not working for campaign_name and camppaign_source-medium report

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Fixed flat report not working for campaign_name and camppaign_source-medium report
Fixes: #PG-3258, #144

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: API.php [hunk 1]
```diff
@@ -25,10 +25,10 @@
  */
 class API extends \Piwik\Plugin\API
 {
-    protected function getDataTable($name, $idSite, $period, $date, $segment, $expanded = false, $idSubtable = null)
+    protected function getDataTable($name, $idSite, $period, $date, $segment, $expanded = false, $idSubtable = null, $flat = false)
```
Comment (sgiehl) [inline API.php]: For consistency reasons I guess I would prefer to use the same order of parameters as for `Archive::createDataTableFromArchive`

```suggestion
    protected function getDataTable($name, $idSite, $period, $date, $segment, $expanded = false, $flat = false, $idSubtable = null)
```

**Review by snake14 — COMMENTED** [review]
The changes look good. I'm not sure if I'm doing something wrong, but I still experience the issue with paging a flat names report when I check out these changes in my local.

**Comment by AltamashShaikh** [comment]
> The changes look good. I'm not sure if I'm doing something wrong, but I still experience the issue with paging a flat names report when I check out these changes in my local.

@snake14 As discussed deploying the code on [demo2](https://demo2.matomo.org/index.php?module=CoreHome&action=index&idSite=82&period=day&date=yesterday#?idSite=82&period=day&date=yesterday&category=Referrers_Referrers&subcategory=Referrers_Campaigns) works, may be something specific in your local?

**Review by snake14 — APPROVED** [review]
Confirmed that the change is working on demo2. So, there's probably just something off with my local instance causing it to not work right.

**Pull request merged**